__tests__/commands/install/resolutions.js

```diff
@@ -115,3 +115,26 @@ test.concurrent('install with resolution settings should correctly bailout durin
     },
   );
 });
+
+test.concurrent('adding resolutions after install should cause lockfile regeneration on second install', (): Promise<
+  void,
+> => {
+  return runInstall(
+    {},
+    {source: 'resolutions', cwd: 'adding-resolutions-should-casue-lockfile-regeneration'},
+    async (config, reporter): Promise<void> => {
+      const packageJson = await fs.readFile(path.join(config.cwd, 'package.json'));
+      // create new package.json with resolutions which override e/left-pad version
+      const newPackageJson = packageJson.replace(/}\n}/g, '  },\n"resolutions": {\n    "e/left-pad": "1.1.1"\n  }\n}');
```

Comment:
Can you use `JSON.parse`, modify the parsed object and then use `JSON.stringify` to write it back instead of using `.replace` on the JSON string?